src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,95 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.User;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+
+    @Before
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void testRegister_ValidUser_Success() throws UserRegistrationException {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user = new User();
+        user.setLogin("username");
+        user.setPassword("password");
+        user.setAge(20);
+        User registeredUser = registrationService.register(user);
+        Assert.assertEquals(user, registeredUser);
+    }
+
+    @Test
+    public void testRegister_UserWithExistingLogin_Failure() {
+        RegistrationService registrationService = new RegistrationServiceImpl();
```

Comment:
Remove the local RegistrationService instance creation and use the class-level instance instead.